### PR TITLE
Proof checker fix

### PR DESCRIPTION
**Pull request by SimonGuilloud**
To merge after #181. Fixes #182

### CODE REVIEW

File: lisa-kernel/src/main/scala/lisa/kernel/proof/SCProofChecker.scala [hunk 1]
```diff
@@ -225,16 +233,20 @@ object SCProofChecker {
...
           case RightIff(b, t1, t2, phi, psi) =>
             val phiImpPsi = ConnectorFormula(Implies, Seq(phi, psi))
             val psiImpPhi = ConnectorFormula(Implies, Seq(psi, phi))
             val phiIffPsi = ConnectorFormula(Iff, Seq(phi, psi))
             if (isSameSet(b.left, ref(t1).left union ref(t2).left))
-              if (isSameSet(b.right + phiImpPsi + psiImpPhi, ref(t1).right union ref(t2).right + phiIffPsi))
+              if (
+                isSubset(ref(t1).right, b.right + phiImpPsi) &&
+                isSubset(ref(t2).right, b.right + psiImpPhi) &&
+                isSubset(b.right, ref(t1).right union ref(t2).right + phiIffPsi)
```
Comment (sankalpgambhir) [inline lisa-kernel/src/main/scala/lisa/kernel/proof/SCProofChecker.scala:248]: I was adapting this to basic step tactics, and noticed that this is perhaps too restrictive? We should have a more uniform policy on where and how much we allow weakening. 

It is perhaps enough to check that $\phi \iff \psi \in bot.right$.

**Comment by sankalpgambhir** [comment]
These changes also need to simultaneously be propagated to the `BasicStepTactic` file. Please take a look at SimonGuilloud/lisa#5 .

File: lisa-kernel/src/main/scala/lisa/kernel/proof/SCProofChecker.scala (hunk 1 above)
Comment (SimonGuilloud) [reply lisa-kernel/src/main/scala/lisa/kernel/proof/SCProofChecker.scala:248]: Maybe, but it's the same semantic as what we had. In general we don't permit arbitrary weakening in non-Weakening proof steps.

**Review by sankalpgambhir — APPROVED** [review]
Thanks for merging my PR. LGTM now. I suppose you'll take care of the merge conflict.

**Comment by SimonGuilloud** [comment]
Thank you :)
Will merge this https://github.com/epfl-lara/lisa/pull/180 first

**Pull request merged**